From f3bbfb31d4394e1f825d7aa3940f12e3b329f846 Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Tue, 19 Jul 2022 17:50:41 +0300
Subject: [PATCH] [SW-96301] gaudi2: erom: fix fw sram base address

FW sram address base is wrong, causing things like ppboot hash
calculation to be incorrect.

Change-Id: I4c12cdb27c6a23389b549c4c681e4e8c20b296c3
Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
---

diff --git a/app/ext_rom/erom_tpm.c b/app/ext_rom/erom_tpm.c
index 7c3a90a..c18d589 100644
--- a/app/ext_rom/erom_tpm.c
+++ b/app/ext_rom/erom_tpm.c
@@ -28,9 +28,10 @@
 #define FLASH_64KB_PADD			65536
 #define FLASH_VALID_MASK		0x1FFFFF
 #define SECUREBOOT_MAX_PPBOOT_IMG_SIZE	46080
-#define HL_FWSRAM_OFFSET		0x7FD0000
+#define HL_FWSRAM_OFFSET		0x3FD0000
+#define HL_GEN_MEM_BASE			0x24000000
 
-#define HL_FWSRAM_BASE			(HL_SDRAM_BASE + \
+#define HL_FWSRAM_BASE			(HL_GEN_MEM_BASE + \
 					 HL_FWSRAM_OFFSET)
 
 static int arcboot_rom_status_read(enum arcboot_rom_progress progress)
